packet_packer.go

```diff
@@ -185,9 +191,24 @@ func (p *packetPacker) composeNextPacket(
 	var payloadLength protocol.ByteCount
 	var payloadFrames []frames.Frame
 
+	// STOP_WAITING and ACK will always fit
 	if p.stopWaiting != nil {
-		p.controlFrames = append(p.controlFrames, p.stopWaiting)
+		payloadFrames = append(payloadFrames, p.stopWaiting)
+		l, err := p.stopWaiting.MinLength(p.version)
+		if err != nil {
+			return nil, err
+		}
+		payloadLength += l
 	}
+	if p.ackFrame != nil {
+		payloadFrames = append(payloadFrames, p.ackFrame)
+		l, err := p.ackFrame.MinLength(p.version)
```

Comment:
Strictly speaking, this never errors for an ACK frame. I don't mind checking for the error here though for now, and removing the error from `MinLength` as soon as we finally drop the STOP_WAITING.